### DISCUSSION TITLE
LDAP(Microsoft ActiveDirectory) Configuration on Kyuubi

### DISCUSSION THREAD

**Discussion by mahalingamagesthian** [apache/kyuubi#6409] [General]
We are looking at setting up LDAP configuration on Kyuubi. We use Microsoft Active Directory. In Kyuubi we have the following LDAP configs:
`kyuubi.authentication=LDAP
    kyuubi.authentication.ldap.baseDN=DC=abc,DC=abdc,DC=org
    kyuubi.authentication.ldap.domain=*****
    kyuubi.authentication.ldap.binddn=username
    kyuubi.authentication.ldap.bindpw=password
    kyuubi.authentication.ldap.url=ldap://msadserver.com:389
   kyuubi.authentication.ldap.customLDAPQuery=(&(objectClass=group)(CN=group1***))`
   
   When we login to Kyuubi we get the following exception:

Could not open client transport with JDBC Uri: jdbc:kyuubi://kyuubi****:10009/;: Peer indicated failure: PLAIN auth failed: javax.security.sasl.AuthenticationException: LDAP Authentication failed for ip56 [Caused by javax.naming.PartialResultException: Unprocessed Continuation Reference(s); remaining name 'DC=abc,DC=abdc,DC=org']

When I research around this, the following are the options to fix the issue,
1. -Djava.naming.referral=follow
2. -Djava.naming.referral=ignore

passing these options as part of KYUUBI_JAVA_OPTS didn't fix this issue ? 

Is there any one who has faced this issue before ? Are there any suggestions to fix this problem ?

**Comment by RusbNoit** [comment]
Kyuubi need add ignore PartialResultException for LDAP Active Directory connection. 
More detail inforfation: https://myshittycode.com/2017/03/26/ldaptemplate-javax-naming-partialresultexception-unprocessed-continuation-references-remaining-name/

You can try one of  two solutions:
1) Use AD global catalog. Change Port from 389, 636 to  3268, 3269 (SSL).
2) Change pattern to OU : kyuubi.authentication.ldap.userDNPattern=sAMAccountName=%s,CN=Users,DC=apache,DC=org
